File: pydss/general/CableDataImpl.py
from dss.general.impl.ConductorDataImpl import (ConductorDataImpl,)
from dss.common.impl.DSSGlobals import (DSSGlobals,)
from dss.general.CableData import (CableData,)
from dss.common.impl.DSSClassDefs import (DSSClassDefs,)
from dss.parser.impl.Parser import (Parser,)


class CableDataImpl(ConductorDataImpl, CableData):
    numCableClassProps = None

    def __init__(self):
        self.numCableClassProps = 4
        self.classType = DSSClassDefs.DSS_OBJECT

    def countProperties(self):
        self.numProperties = self.numProperties + self.numCableClassProps
        super(CableDataImpl, self).countProperties()

    def defineProperties(self):
        self.propertyName[self.activeProperty + 1] = 'EpsR'
        self.propertyName[self.activeProperty + 2] = 'InsLayer'
        self.propertyName[self.activeProperty + 3] = 'DiaIns'
        self.propertyName[self.activeProperty + 4] = 'DiaCable'
        self.propertyHelp[self.activeProperty + 1] = 'Insulation layer relative permittivity; default is 2.3.'
        self.propertyHelp[self.activeProperty + 2] = 'Insulation layer thickness; same units as radius; no default. ' + 'With DiaIns, establishes inner radius for capacitance calculation.'
        self.propertyHelp[self.activeProperty + 3] = 'Diameter over insulation layer; same units as radius; no default.' + 'Establishes outer radius for capacitance calculation.'
        self.propertyHelp[self.activeProperty + 4] = 'Diameter over cable; same units as radius; no default.'
        self.activeProperty = self.activeProperty + self.numCableClassProps
        super(CableDataImpl, self).defineProperties()

    def classEdit(self, activeObj, paramPointer):
        parser = Parser.getInstance()
        result = 0
        # continue parsing with contents of parser
        if paramPointer >= 0:
            cd = activeObj
            _0 = paramPointer
            _1 = False
            while True:
                if _0 == 0:
                    _1 = True
                    cd.setEpsR(parser.makeDouble())
                    break
                if (_1 is True) or (_0 == 1):
                    _1 = True
                    cd.setInsLayer(parser.makeDouble())
                    break
                if (_1 is True) or (_0 == 2):
                    _1 = True
                    cd.setDiaIns(parser.makeDouble())
                    break
                if (_1 is True) or (_0 == 3):
                    _1 = True
                    cd.setDiaCable(parser.makeDouble())
                    break
                if True:
                    _1 = True
                    super(CableDataImpl, self).classEdit(activeObj, paramPointer - self.numCableClassProps)
                    break
                break
            # Check for critical errors
            _2 = paramPointer
            _3 = False
            while True:
                if _2 == 0:
                    _3 = True
                    if cd.getEpsR() < 1.0:
                        DSSGlobals.doSimpleMsg('Error: Insulation permittivity must be greater than one for CableData ' + cd.getName(), 999)
                    break
                if (_3 is True) or (_2 == 1):
                    _3 = True
                    if cd.getInsLayer() <= 0.0:
                        DSSGlobals.doSimpleMsg('Error: Insulation layer thickness must be positive for CableData ' + cd.getName(), 999)
                    break
                if (_3 is True) or (_2 == 2):
                    _3 = True
                    if cd.getDiaIns() <= 0.0:
                        DSSGlobals.doSimpleMsg('Error: Diameter over insulation layer must be positive for CableData ' + cd.getName(), 999)
                    break
                if (_3 is True) or (_2 == 3):
                    _3 = True
                    if cd.getDiaCable() <= 0.0:
                        DSSGlobals.doSimpleMsg('Error: Diameter over cable must be positive for CableData ' + cd.getName(), 999)
                    break
                break
        return result

    def classMakeLike(self, otherObj):
        otherCableData = otherObj
        cd = DSSGlobals.activeDSSObject
        cd.setEpsR(otherCableData.getEpsR())
        cd.setInsLayer(otherCableData.getInsLayer())
        cd.setDiaIns(otherCableData.getDiaIns())
        cd.setDiaCable(otherCableData.getDiaCable())
        super(CableDataImpl, self).classMakeLike(otherObj)

    def setNumCableClassProps(self, num):
        self.numCableClassProps = num

    def getNumCableClassProps(self):
        return self.numCableClassProps

File: core/dss-core/src/main/java/com/ncond/dss/general/impl/CableDataImpl.java
package com.ncond.dss.general.impl;

import com.ncond.dss.common.impl.DSSClassDefs;
import com.ncond.dss.common.impl.DSS;
import com.ncond.dss.general.CableData;
import com.ncond.dss.general.CableDataObj;
import com.ncond.dss.general.DSSObject;
import com.ncond.dss.parser.impl.Parser;

public class CableDataImpl extends ConductorDataImpl implements CableData {

	private int numCableClassProps;

	public CableDataImpl() {
		numCableClassProps = 4;
		classType = DSSClassDefs.DSS_OBJECT;
	}

	protected void countProperties() {
		numProperties = numProperties + numCableClassProps;
		super.countProperties();
	}

	protected void defineProperties() {
		propertyName[activeProperty + 1] = "EpsR";
		propertyName[activeProperty + 2] = "InsLayer";
		propertyName[activeProperty + 3] = "DiaIns";
		propertyName[activeProperty + 4] = "DiaCable";

		propertyHelp[activeProperty + 1] = "Insulation layer relative permittivity; default is 2.3.";
		propertyHelp[activeProperty + 2] = "Insulation layer thickness; same units as radius; no default. " +
				"With DiaIns, establishes inner radius for capacitance calculation.";
		propertyHelp[activeProperty + 3] = "Diameter over insulation layer; same units as radius; no default." +
				"Establishes outer radius for capacitance calculation.";
		propertyHelp[activeProperty + 4] = "Diameter over cable; same units as radius; no default.";

		activeProperty = activeProperty + numCableClassProps;
		super.defineProperties();
	}

	protected int classEdit(DSSObject activeObj, int paramPointer) {
		Parser parser = Parser.getInstance();

		int result = 0;
		// continue parsing with contents of parser
		if (paramPointer >= 0) {
			CableDataObj cd = (CableDataObj) activeObj;
			switch (paramPointer) {
			case 0:
				cd.setEpsR(parser.makeDouble());
				break;
			case 1:
				cd.setInsLayer(parser.makeDouble());
				break;
			case 2:
				cd.setDiaIns(parser.makeDouble());
				break;
			case 3:
				cd.setDiaCable(parser.makeDouble());
				break;
			default:
				super.classEdit(activeObj, paramPointer - numCableClassProps);
				break;
			}
			/* Check for critical errors */
			switch (paramPointer) {
			case 0:
				if (cd.getEpsR() < 1.0)
					DSS.doSimpleMsg("Error: Insulation permittivity must be greater than one for CableData " + cd.getName(), 999);
				break;
			case 1:
				if (cd.getInsLayer() <= 0.0)
					DSS.doSimpleMsg("Error: Insulation layer thickness must be positive for CableData " + cd.getName(), 999);
				break;
			case 2:
				if (cd.getDiaIns() <= 0.0)
					DSS.doSimpleMsg("Error: Diameter over insulation layer must be positive for CableData " + cd.getName(), 999);
				break;
			case 3:
				if (cd.getDiaCable() <= 0.0)
					DSS.doSimpleMsg("Error: Diameter over cable must be positive for CableData " + cd.getName(), 999);
				break;
			}
		}

		return result;
	}

	protected void classMakeLike(DSSObject otherObj) {

		CableDataObj otherCableData = (CableDataObj) otherObj;
		CableDataObj cd = (CableDataObj) DSS.activeDSSObject;

		cd.setEpsR(otherCableData.getEpsR());
		cd.setInsLayer(otherCableData.getInsLayer());
		cd.setDiaIns(otherCableData.getDiaIns());
		cd.setDiaCable(otherCableData.getDiaCable());

		super.classMakeLike(otherObj);
	}

	public void setNumCableClassProps(int num) {
		numCableClassProps = num;
	}

	public int getNumCableClassProps() {
		return numCableClassProps;
	}

}
